docs/en/setup/envoy/als_setting.md

```diff
@@ -29,8 +29,9 @@ On Istio version 1.6.0+, if Istio is installed with [`demo` profile](https://ist
     
 - Activate SkyWalking [Envoy Receiver](../backend/backend-receivers.md). This is activated by default. 
 
-- Choose an ALS analyzer. There are two available analyzers, `k8s-mesh` and `mx-mesh`.
-  Set the system environment variable **SW_ENVOY_METRIC_ALS_HTTP_ANALYSIS** such as `SW_ENVOY_METRIC_ALS_HTTP_ANALYSIS=k8s-mesh`
+- Choose an ALS analyzer. There are two available analyzers, `k8s-mesh` and `mx-mesh` for both HTTP access logs and TCP access logs.
+  Set the system environment variable **SW_ENVOY_METRIC_ALS_HTTP_ANALYSIS** and **SW_ENVOY_METRIC_ALS_TCP_ANALYSIS**
+  such as `SW_ENVOY_METRIC_ALS_HTTP_ANALYSIS=k8s-mesh`, `SW_ENVOY_METRIC_ALS_TCP_ANALYSIS=k8s-mesh`
```

Comment:
So, TCP traffic is not working on VM. Correct?

Reply:
> So, TCP traffic is not working on VM. Correct?

I think it should work on VM, from the code implementation I don’t see any difference between VM and Kubernetes pods in terms of TCP traffic

Reply:
How you get the VM node info from IP?

Reply:
> How you get the VM node info from IP?

You should use `mx-mesh` for VM scenario, we support both `mx-mesh` and `k8s-mesh` for TCP access logs, the doc is only a demonstration.

Reply:
meta exchange works in HTTP 1/2 due to the HTTP 1/2 has a header to propagate the metadata transparently. I remember in the Istio doc, I shared with you, it says, using some mTLS field to do that, but whether it is implemented? I think you need another round of testing to confirm, we can't be sure.

Reply:
> meta exchange works in HTTP 1/2 due to the HTTP 1/2 has a header to propagate the metadata transparently. I remember in the Istio doc, I shared with you, it says, using some mTLS field to do that, but whether it is implemented? I think you need another round of testing to confirm, we can't be sure.

According to the tests result (I’ve added tcp topology verification with mongodb enabled in the test env), it seems to passed.

Reply:
> According to the tests result (I’ve added tcp topology verification with mongodb enabled in the test env), it seems to passed.

Doesn't that run on k8s?

Reply:
> > According to the tests result (I’ve added tcp topology verification with mongodb enabled in the test env), it seems to passed.
> 
> Doesn't that run on k8s?

Well. They run on Kubernetes but they still use TCP protocol to communicate, and we still use mx analyzer to analyze the logs. Your concern is only about that TCP protocol cannot carry metadata transparently, right?

Reply:
Yes, but if you are on k8s, IP mapping works, doesn't it?

Reply:
> Yes, but if you are on k8s, IP mapping works, doesn't it?

The test case only tests an analyzer at a time, when it is testing mx mesh, k8s mesh is disabled

Reply:
We need to be sure, whether meta exchange really works for TCP.

Reply:
https://docs.google.com/document/d/1s6ou__qRL4UiWY1amyk5p8YzOMQvBhfR15o1U_ZLIow/edit#heading=h.qex63c29z2to
According to this, have the demo activated the mTLS between service and MongoDB?